### ISSUE TITLE
By far the best replacement I've found for Clean Links.

### ISSUE THREAD

**Issue by ImSpecial** [sblask/firefox-skip-redirect#21]
I've tried a lot of these url "cleaners" and this one just works! Finally I can drop the now dead add-on Clean Links (R.I.P) and use Firefox with e10s enabled. So thank you for all your hard work Mr. Dev, it's appreciated.

Now for some constructive criticism (but don't worry, it's not much).

I've tried both the stable version 1.0.7 and the dev build 2.0.0b5, and between the two I like a few things about 1.0.7 that are _different_ in the way they work now in the beta, I'm not sure if they are "coming later" because as I understand it you are working on making the Web Extension version.

Anyway with the Beta version I am not a fan of the [white box](http://i.imgur.com/QrLtvYH.png) that pops up with no way of disabling it, I much preferred the old method where you just hovered over the icon button, it did a quick flash saying Skip, and showed the before and after URL, it was clean and simple. If that is no longer possible, I'd rather just disable that popup box altogether.

Another thing I miss is the quick way to toggle off the add-on, before it was a simple click which made the icon fade grey to show it was disabled, but now it pops up a huge white window (lots of dead white space btw) and to make matters worse you need to hit save which requires scrolling down that window.

Now the most important one, basically unusable until this is fixed. The icon seems a bit _low_ and could be raised a pixel or two to better align it with other toolbar icons. https://i.imgur.com/mcsXCPp.png

Basically other then those listed above, the 2.0.0b5 build works great and even gets stuff that the stable version misses. Great job once again on this add-on.

**Comment by sblask** [comment]
Thanks!

I actually prefer the notification because I missed the little red `skip` way to often and wondered why some sites didn't work. But I can make this configurable...

Toggle vs options popup: For me, I got a lot of problems due to redirect skipping(that's why I added the whitelist mode...) so having the options more accessible was important to me. But I can make this configurable too. Or maybe it's enough to just save once the selection changes? There is much less white space in the latest version but it's only available on Chrome as of today, I've got an older version that has been waiting for review for ages already and I can't put the latest version up yet because of data migration...

I can change the icon, I have no real opinion about "where" exactly the arrow should be...

As I said, reviews take forever and then I need to find the time to make the changes, so don't hold your breath...

**Comment by ImSpecial** [comment]
> I actually prefer the notification because I missed the little red skip way to often and wondered why some sites didn't work. But I can make this configurable...

I'm fine with configurable, personally for me, if something like this is working, I don't want to be bothered by notifications, I want it to be as invisible as possible, web pages breaking is rare, at least for my usage and since I am the one that installed this add-on I kinda know what's going on when a web page 404's or doesn't load correctly when I click on a link, so the first thing that goes into my mind is "hmm maybe this add-on" is causing this. Another thing I noticed with this popup was I use multiple monitors and sometimes Firefox is on my second one, but even so the pop up was always showing on my main monitor.

> Toggle vs options popup: For me, I got a lot of problems due to redirect skipping(that's why I added the whitelist mode...) so having the options more accessible was important to me. But I can make this configurable too. Or maybe it's enough to just save once the selection changes? There is much less white space in the latest version but it's only available on Chrome as of today, I've got an older version that has been waiting for review for ages already and I can't put the latest version up yet because of data migration...

I'm fine with either or, the options such as whitelisting are nice, don't get rid of them, I'm only suggesting a quicker way of toggling on/off in case I need to click a single link a single time that's not working correctly, I'm just not a fan of it taking up all that space and having the scroll down to hit save, but as you said there is less space used in a newer version so maybe my gripe is already solved, or if it auto saved when something changed then that would make things indeed quicker.

Icon is hardly a big deal, I just sorta noticed it was off center compared to my other toolbar icons so it was my OCD that made me post about it :P, it's whatever though.

Thanks for your time and I'll try not to hold my breath.

**Comment by sblask** [comment]
I just uploaded a new version with configurable notification, skip badge, tooltip and auto-save in options. Now it just has to go through review...

**Comment by sblask** [comment]
If you can't wait, I also uploaded a beta version (see Development Channel at the bottom of the add-on page) that you can install directly...

**Comment by ImSpecial** [comment]
Good job dude! I've only used it a tiny bit, but it seems pretty darn perfect right now. I checked on AMO and there is the 2.1.5 version and 2.1.5b dev version, are they different?

**Comment by sblask** [comment]
No, they aren't. I just added the beta with the same content because it doesn't require a review. The first review took nearly a month so this is what I expected, but this one took only half a day. So I guess I can close this now.
